src/components/Item/Item.component.jsx

```diff
@@ -9,19 +9,40 @@ const Item = props => {
   const itemId = props.id.toString();
   const localToken = props.localToken;
   const over24 = props.over24;
+  const resupplyPeriod = props.resupplyPeriod;
+  let lastEstimate = props.lastEstimate;
+  let latestInterval = props.latestInterval;
+  let numberOfPurchases = props.numberOfPurchases;
+  let nextPurchaseInterval = props.nextPurchaseInterval;
+  let lastPurchaseDate = props.date;
+  let lastPurchasedTimeInSeconds = 0;
 
-  useEffect(() => {
-    if (over24 === false) {
-      document.getElementById(itemId).setAttribute('class', 'item__hightlight');
-    }
-  });
-
-  //To update the purchase date
+  // To update the purchase date
   const markPurchased = event => {
     event.preventDefault();
     const date = new Date();
 
-    document.getElementById(itemId).setAttribute('class', 'highlight');
+    // To update the number of purchases
+    numberOfPurchases += 1;
+
+    // To update the latest interval
+    if (lastPurchaseDate === null) {
+      latestInterval = 0;
+    } else {
+      let currentTimeInSeconds = new Date().getTime() / 1000;
+      lastPurchasedTimeInSeconds = lastPurchaseDate.seconds;
+      latestInterval = Math.ceil(
+        (currentTimeInSeconds - lastPurchasedTimeInSeconds) / 86400,
```

Comment:
This code seems duplicative of code already merged. See here. https://github.com/the-collab-lab/tcl-11-smart-shopping-list/blob/47adf15c7d3f42097ba8f16262b59eaedeb2ae0e/src/components/Item/Item.component.jsx#L41